hi-ml-cpath/testhisto/testhisto/datamodules/test_histo_datamodule.py

```diff
@@ -0,0 +1,173 @@
+        sample = next(iter(dataloader))
+        assert len(sample[SlideKey.IMAGE]) == expected_batch_size
+
+    _assert_batch_size_matching(datamodule.train_dataloader(), batch_size)
+    _assert_batch_size_matching(datamodule.val_dataloader(), batch_size_inf)
+    _assert_batch_size_matching(datamodule.test_dataloader(), batch_size_inf)
+
+
+@pytest.mark.skipif(no_gpu, reason="Test requires GPU")
+@pytest.mark.gpu
+@pytest.mark.parametrize("max_bag_size, max_bag_size_inf", [(2, 0), (2, 3)])
+def test_slides_datamodule_different_bag_sizes(
+    mock_panda_slides_root_dir: Path, max_bag_size: int, max_bag_size_inf: int
+) -> None:
+    datamodule = PandaSlidesDataModule(
+        root_path=mock_panda_slides_root_dir,
+        batch_size=2,
+        max_bag_size=max_bag_size,
+        max_bag_size_inf=max_bag_size_inf,
+        tile_size=28,
+        level=0,
+    )
+    # To account for the fact that slides datamodule fomats 0 to None so that it's compatible with TileOnGrid transform
+    max_bag_size_inf = max_bag_size_inf if max_bag_size_inf != 0 else None  # type: ignore
+    _assert_correct_bag_sizes(datamodule, max_bag_size, max_bag_size_inf, [4, 4])
+
+
+@pytest.mark.parametrize("max_bag_size, max_bag_size_inf", [(2, 0), (2, 3)])
+def test_tiles_datamodule_different_bag_sizes(
+    mock_panda_tiles_root_dir: Path, max_bag_size: int, max_bag_size_inf: int
+) -> None:
+    datamodule = PandaTilesDataModule(
+        root_path=mock_panda_tiles_root_dir,
+        batch_size=2,
+        max_bag_size=max_bag_size,
+        max_bag_size_inf=max_bag_size_inf,
+    )
+    _assert_correct_bag_sizes(datamodule, max_bag_size, max_bag_size_inf, max_expected_bag_sizes=[4, 5])
+
+
+@pytest.mark.skipif(no_gpu, reason="Test requires GPU")
+@pytest.mark.gpu
+@pytest.mark.parametrize("batch_size, batch_size_inf", [(2, 2), (2, 1), (2, None)])
+def test_slides_datamodule_different_batch_sizes(
+    mock_panda_slides_root_dir: Path, batch_size: int, batch_size_inf: Optional[int],
+) -> None:
+    datamodule = PandaSlidesDataModule(
+        root_path=mock_panda_slides_root_dir,
+        batch_size=batch_size,
+        batch_size_inf=batch_size_inf,
+        max_bag_size=16,
+        max_bag_size_inf=16,
+        tile_size=28,
+        level=0,
+    )
+    _assert_correct_batch_sizes(datamodule, batch_size, batch_size_inf)
+
+
+@pytest.mark.parametrize("batch_size, batch_size_inf", [(2, 2), (2, 1), (2, None)])
+def test_tiles_datamodule_different_batch_sizes(
+    mock_panda_tiles_root_dir: Path, batch_size: int, batch_size_inf: Optional[int],
+) -> None:
+    datamodule = PandaTilesDataModule(
+        root_path=mock_panda_tiles_root_dir,
+        batch_size=batch_size,
+        batch_size_inf=batch_size_inf,
+        max_bag_size=16,
+        max_bag_size_inf=16,
+    )
+    _assert_correct_batch_sizes(datamodule, batch_size, batch_size_inf)
+
+
+def _assert_sampler_state(datamodule: HistoDataModule, stages: List[ModelKey], expected_none: bool) -> None:
```

Comment:
[Just for awareness] "Assert X" normally refers to a Boolean predicate, meaning "verify that X is true". For any other uses, something like "validate" or "check" would sound more natural.

Reply:
In this particular case, you could be more explicit, like `_validate_sampler_type()`.